### PR TITLE
Upgrade to Asciidoctor.js 1.5.6

### LINKED ISSUE #205: include from 2nd level asc file via tags is broken in v2.7.0
## Description

Indirect code inclusion from 2nd level asc file stopped working in v2.7.0.  This worked in v2.6.0 . For some reason it now looks in the atom install directory for the python file.

- Atom version: 1.12.6
- OS: Windows 10
- `asciidoc-preview` version: 2.7.0
- `language-asciidoc` version: 1.6.2

## Syntax example
```
├── parent.asc
├── 2nd.asc
└── subdir\foo.py
```
## parent.asc
```adoc
include::2nd.asc[tags=test]
```
## 2nd.asc

```adoc
tag::test[]
[source,python]
-----
include::subdir/foo.py[tags=bar]
-----
end::test[]
```

## Error message

Asciidoctor.js error
```
Rendering error
Please verify your document syntax.

Details: Error: ENOENT: no such file or directory, open 'C:\Users\elefant\AppData\Local\atom\app-1.12.6\subdir\foo.py'

[code: ENOENT, errno: -4058, syscall: open]

Error: ENOENT: no such file or directory, open 'C:\Users\elefant\AppData\Local\atom\app-1.12.6\subdir\foo.py' 
at Error (native) 
at Object.fs.openSync (fs.js:640:18) 
at Object.module.(anonymous function) [as openSync] (ELECTRON_ASAR.js:168:20) 
at $File.ːinitialize (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:20294:40) 
at singleton_class_alloc.TMP_4 [as $new] (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\opal-runtime\src\opal.js:3139:23) 
at $PreprocessorReader.ːopen (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:20254:45) 
at $PreprocessorReader.ːpreprocess_include (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:31659:141) 
at $PreprocessorReader.ːprocess_line (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:31380:36) 
at $PreprocessorReader.ːpeek_line (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:30986:41) 
at $PreprocessorReader.ːpeek_line (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:31411:129) 
at $PreprocessorReader.TMP_6 [as $has_more_lines?] (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:30961:98) 
at $PreprocessorReader.ːread_line (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:31030:189) 
at $PreprocessorReader.ːread_lines_until (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:31193:102) 
at singleton_class_alloc.ːbuild_block (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:29202:26) 
at singleton_class_alloc.ːnext_block (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:29009:24) 
at singleton_class_alloc.ːnext_section (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:28642:40) 
at singleton_class_alloc.ːnext_section (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:28633:23) 
at singleton_class_alloc.ːparse (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:28479:21) 
at $Document.ːparse (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:27419:32) 
at module_constructor.ːload (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:33226:65) 
at module_constructor.ːconvert (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:33306:20) 
at module_constructor.Opal.Asciidoctor.$$class.$$proto.convert (C:\Users\elefant\.atom\packages\asciidoc-preview\node_modules\asciidoctor.js\dist\asciidoctor.js:33455:21) 
at Object.module.exports (file:///C:/Users/elefant/.atom/packages/asciidoc-preview/lib/worker.coffee:34:24) 
at process. (C:\Users\elefant\AppData\Local\atom\app-1.12.6\resources\app.asar\src\task-bootstrap.js:84:10) at emitTwo (events.js:106:13) 
at process.emit (events.js:191:7) at process.nextTick (internal/child_process.js:744:12) at _combinedTickCallback (internal/process/next_tick.js:67:7) at process._tickCallback (internal/process/next_tick.js:98:9)
```

### PR DESCRIPTION
**Pull request by Mogztter**
## Description

Upgrade to the latest release of Asciidoctor.js: https://github.com/asciidoctor/asciidoctor.js/releases/tag/v1.5.6-rc.1

Should resolve #205

### CODE REVIEW

**Comment by ldez** [comment]
wrong content.
You must apply your change on the master.

File: package.json [hunk 1]
```diff
@@ -165,45 +165,15 @@
       "collapsed": true,
       "properties": {
         "enabled": {
-          "title": "Enable PDF export using Asciidoctor PDF.",
-          "description": "The [asciidoctor-pdf](https://github.com/asciidoctor/asciidoctor-pdf) command must be available on your `PATH`.",
+          "title": "Use asciidoctor-pdf",
```
Comment (ldez) [inline package.json]: bad rebase

**Comment by Mogztter** [comment]
Yes my branch is out of sync, I also need to upgrade the code (ie. to apply https://github.com/asciidoctor/atom-asciidoc-preview/pull/202).

You can ignore this pull request for now :unamused:

**Comment by Mogztter** [comment]
@ldez Should be OK now

File: lib/worker.coffee [hunk 2]
```diff
@@ -59,10 +60,7 @@ registerBlockPositions = (doc) ->
       doc.id = "__asciidoctor-preview-#{doc.$object_id()}__"
     doc.header.id = doc.id
 
-  # Use Ruby API:
-  # because `doc.findBy()` doesn't yet accept a filter function as parameter.
-  # https://github.com/asciidoctor/asciidoctor.js/issues/282
-  blocks = Opal.block_send doc, 'find_by', (b) -> b.$lineno() isnt Opal.nil
+  blocks = doc.findBy((b) -> b.$lineno() is 'number')
```
Comment (Mogztter) [inline lib/worker.coffee]: @ldez Once https://github.com/asciidoctor/asciidoctor.js/pull/297 is merged, I think we could use the syntax `?`:

```javascript
blocks = doc.findBy((b) -> b.getLineNumber()?)
```

**Comment by ldez** [comment]
@Mogztter Maybee you can update this PR with the latest version?

**Comment by Mogztter** [comment]
@ldez Asciidoctor.js 1.5.6 (based on Asciidoctor 1.5.6.1) is not yet ready, some path resolution are not working at the moment.
I've updated this pull request to use Asciidoctor.js 1.5.5. I've also updated the tests because `object_id` generation has changed.

Is there anything not working with this version ? I vaguely remember some issues with includes... ?

File: lib/worker.coffee [hunk 3]
```diff
@@ -59,10 +60,7 @@ registerBlockPositions = (doc) ->
       doc.id = "__asciidoctor-preview-#{doc.$object_id()}__"
```
Comment (Mogztter) [inline lib/worker.coffee]: @mojavelinux I'm wondering if there's a more stable (unique) identifier than `object_id` ?

**Comment by Mogztter** [comment]
Nested include are working (again) on Windows:
nested-include

`object_id` generation changed again, I will update the tests.

File: lib/worker.coffee (hunk 3 above)
Comment (mojavelinux) [reply lib/worker.coffee]: Not at the moment, though this is not the first request for it. We'll need an upstream issue to discuss.

**Comment by Mogztter** [comment]
I'm confused... `object_id` does not return the same value between Windows (Appveyor) and Linux (Travis) :confused: 

Also I think that `lineno` are wrong on nested list (probably related to https://github.com/asciidoctor/asciidoctor/pull/2547). See the comment in spec.

File: spec/worker-blocks-positions-spec.coffee [hunk 4]
```diff
@@ -64,11 +64,13 @@ describe 'worker when the scroll mode is activated', ->
...
     runs =>
       expect(Object.keys @blockPositions).toHaveLength 5
-      expect(@blockPositions[1]).toBe '__asciidoctor-preview-420__'
+      expect(@blockPositions[1]).toBe '__asciidoctor-preview-1554__'
       expect(@blockPositions[3]).toBe '_first_section'
-      expect(@blockPositions[6]).toBe '__asciidoctor-preview-468__'
-      expect(@blockPositions[7]).toBe '__asciidoctor-preview-492__'
-      expect(@blockPositions[8]).toBe '__asciidoctor-preview-512__'
+      # line number on nested list is wrong
+      # maybe related to https://github.com/asciidoctor/asciidoctor/pull/2547
```
Comment (Mogztter) [inline spec/worker-blocks-positions-spec.coffee]: @mojavelinux probably related to https://github.com/asciidoctor/asciidoctor/pull/2547 right ?

Comment (mojavelinux) [reply spec/worker-blocks-positions-spec.coffee]: That's correct.

**Comment by mojavelinux** [comment]
> I'm confused... object_id does not return the same value between Windows (Appveyor) and Linux (Travis)

That's not unexpected. object_id is roughly the same as the serialVersionUID in Java. It's not a stable value. I would not rely on it from one run to the next. You might be better off relying on position in AST.

**Comment by Mogztter** [comment]
> That's not unexpected. object_id is roughly the same as the serialVersionUID in Java. It's not a stable value. I would not rely on it from one run to the next. You might be better off relying on position in AST.

For now I'm just using a simple incrementing number. I don't know if we need more than that ?

**Comment by Mogztter** [comment]
:tada: :tada: :tada:

**Pull request merged**